### PR TITLE
feat(form-field, input): gh-form-field & ghInput

### PR DESCRIPTION
**Pull request by thomaspink**
Issue form field: #50     
Issue input: #28

### CODE REVIEW

File: src/demo-app/input/input-demo.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+<p>Current value: {{emailValue}}</p>
+<p>Touched: {{ emailControl.touched }}</p>
+<p>Dirty: {{ emailControl.dirty }}</p>
+<p>Status: {{ emailControl.control?.status }}</p>
+<div>
+  <gh-form-field>
+    <gh-label>E-Mail:</gh-label>
+    <input ghInput
+      required
+      type="number"
```
Comment (ffriedl89) [inline src/demo-app/input/input-demo.html]: type email

File: src/lib/form-field/form-field.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  border-bottom-right-radius: 0;
+}
+
+.gh-form-field-hint-wrapper {
+  display: flex;
+  padding-top: 6px;
+}
+
+.gh-form-field-hint-spacer {
+  flex: 1 0 1em;
```
Comment (ffriedl89) [inline src/lib/form-field/form-field.scss]: why em here?

File: src/lib/form-field/form-field.scss [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  font-weight: 400;
+  text-decoration: none;
+  appearance: none;
+  display: block;
+  box-sizing: border-box;
+  white-space: normal;
+}
+
+.gh-hint {
+  color: #898989;
```
Comment (ffriedl89) [inline src/lib/form-field/form-field.scss]: is there a color variable for this?

File: src/lib/form-field/form-field.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    this._changeDetectorRef.detectChanges();
+  }
+
+  /** Determines whether a class from the NgControl should be forwarded to the host element. */
+  _shouldForward(prop: string): boolean {
+    let ngControl = this._control ? this._control.ngControl : null;
+    return ngControl && (ngControl as any)[prop];
+  }
+
+  /** Determines whether to display hints or errors. */
```
Comment (ffriedl89) [inline src/lib/form-field/form-field.ts]: More like 'returns if there are errors displayed'

File: src/lib/form-field/form-field.scss [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  padding-top: 6px;
+}
+
+.gh-form-field-hint-spacer {
+  flex: 1 0 1em;
+}
+
+.gh-hint,
+.gh-error {
+  font-family: 'BerninaSans', 'Open Sans', sans-serif;
```
Comment (ffriedl89) [inline src/lib/form-field/form-field.scss]: font centralised

File: src/lib/input/input.spec.ts [hunk 6]
```diff
@@ -0,0 +1,827 @@
...
+    const inputElement: HTMLInputElement =
+      fixture.debugElement.query(By.css('input')).nativeElement;
+    const labelElement: HTMLInputElement =
+      fixture.debugElement.query(By.css('label')).nativeElement;
+
+    expect(inputElement.id).toBe('test-id');
+    expect(labelElement.getAttribute('for')).toBe('test-id');
+  }));
+
+  // TODO @thomaspink: Add when testing files have been added
```
Comment (ffriedl89) [inline src/lib/input/input.spec.ts]: are they added now? ;)

Comment (thomaspink) [reply src/lib/input/input.spec.ts]: not yet. will update it when they are. Issue here: https://github.com/angular/material2/issues/9839

File: src/lib/input/input.ts [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+  /** Whether the element is readonly. */
+  @Input()
+  get readonly(): boolean { return this._readonly; }
+  set readonly(value: boolean) { this._readonly = coerceBooleanProperty(value); }
+  private _readonly = false;
+
+  /** An object used to control when error messages are shown. */
+  @Input() errorStateMatcher: ErrorStateMatcher;
+
+  /** Implemented as part of MatFormFieldControl.*/
```
Comment (ffriedl89) [inline src/lib/input/input.ts]: Mat -> Gh

**Pull request merged**